go/vt/vttablet/tabletmanager/vreplication/controller_plan.go

```diff
@@ -228,12 +254,23 @@ func buildDeletePlan(del *sqlparser.Delete) (*controllerPlan, error) {
 }
 
 func buildSelectPlan(sel *sqlparser.Select) (*controllerPlan, error) {
-	switch sqlparser.ToString(sel.From) {
+	tableExpr, ok := sel.From[0].(*sqlparser.AliasedTableExpr)
```

Comment:
I'm just thinking, could there be a scenario where `From` is an empty slice? e.g. `FROM DUAL` gets truncated? I know we won't be running `FROM DUAL` in `vreplication`; but I'd feel safer if we tested for the length of the slice.

Reply:
Nice catch! Yes, will add some protection to this and any similar cases in these build functions.